src/ess/reduce/widgets/_switchable_widget.py

```diff
@@ -0,0 +1,42 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+from typing import Any
+
+from ipywidgets import Checkbox, HBox, Label, Stack, Widget
+
+from ._config import default_style
+
+
+class SwitchWidget(HBox):
+    """Wrapper widget to handle switchable widgets.
+
+    When you retrieve the value of this widget,
+    it will return the value of the wrapped widget.
+    It is expected not to be set in the workflow if ``enabled`` is False.
+    """
+
+    def __init__(self, wrapped: Widget, name: str = '') -> None:
+        super().__init__()
+        self.enable_box = Checkbox(description='Use Parameter', style=default_style)
```

Comment:
Can we have no text instead, and grey-out/disable the widget instead? Some widgets (like `BinEdgeWidget`) might be quite wide, so saving space would be good.

Reply:
If there's no text, it's hard to tell what's the check box is about.

And I couldn't find how to `grey-out` widgets without using java script api... 
And `disable` is not a common property so it's also not an option.

Reply:
Actually, it's not that bad without the description.

https://github.com/user-attachments/assets/f256927e-49d7-45db-a56d-377d70dc0d73